warzone/Strategy.cpp

```diff
@@ -0,0 +1,281 @@
+#include "Strategy.h";
+
+/// <summary>
+/// Default constructor for PlayerStrategy
+/// </summary>
+PlayerStrategy::PlayerStrategy()
+{
+
+}
+
+/// <summary>
+/// Copy constructor that does nothing since PlayerStrategy has no member variables.
+/// </summary>
+PlayerStrategy::PlayerStrategy(const PlayerStrategy& other)
+{
+
+}
+
+/// <summary>
+/// Assignment operator that does nothing since PlayerStrategy has no member variables.
+/// </summary>
+PlayerStrategy& PlayerStrategy::operator=(const PlayerStrategy& other)
+{
+	return *this;
+}
+
+/// <summary>
+/// Outputs "PlayerStrategy" to the console.
+/// </summary>
+ostream& operator<<(ostream& output, const PlayerStrategy& player)
+{
+	return output << "PlayerStrategy";
+}
+
+/// <summary>
+/// Default constructor for DefaultStrategy
+/// </summary>
+DefaultStrategy::DefaultStrategy()
+{
+
+}
+
+/// <summary>
+/// Copy constructor that does nothing since DefaultStrategy has no member variables.
+/// </summary>
+DefaultStrategy::DefaultStrategy(const DefaultStrategy& other)
+{
+	
+}
+
+/// <summary>
+/// Assignment operator that does nothing since DefaultStrategy has no member variables.
+/// </summary>
+DefaultStrategy& DefaultStrategy::operator=(const DefaultStrategy& other)
+{
+	return *this;
+}
+
+/// <summary>
+/// Prints "DefaultStrategy" to the console
+/// </summary>
+ostream& operator<<(ostream& output, const DefaultStrategy& other)
+{
+	return output << "DefaultStrategy";
+}
+
+/// <summary>
+/// Compares two territories, i and j, based on lexicographical/alphabetical order.
+/// Returns True if the first territory is smaller than the second.
+/// </summary>
+bool compareTerritoriesAlphabetically(Territory* i, Territory* j) {
+	return i->getName().compare(j->getName()) < 0;
+}
+
+/// <summary>
+/// Returns the list of territories to defend. This implementation returns the player's
+/// territories sorted by lexicographical order.
+/// </summary>
+const vector<Territory*>* DefaultStrategy::toDefend(Player* player)
+{
+	std::sort(player->getTerritories()->begin(), player->getTerritories()->end(), compareTerritoriesAlphabetically);
+	return player->getTerritories();
+}
+
+/// <summary>
+/// Returns the list of territories to attack. This implementation returns all territories
+/// adjacent to owned territories that belong to other players, sorted alphabetically.
+/// </summary>
+const vector<Territory*> DefaultStrategy::toAttack(Player* player)
+{
+	// TODO make sure this gets destructed at some point... or return by value
```

Comment:
I see you are returning it, is the TODO still relevant?

Reply:
Yes the todo is still relevant, i still think this is a memory leak that i might look at fixing later.